### ISSUE TITLE
 [Machine learning] Getting started with Gradient Descent algorithm

### ISSUE THREAD

**Issue by jacksonjate1** [section-engineering-education/engineering-education#4138]
#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximize the chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

#### Proposed title of the article
Getting started with Gradient Descent algorithm

Use keyword research to improve your article's SEO.

#### Proposed article introduction
Gradient descent is an optimization algorithm that is used when training a machine learning model. It's based on a convex function and tweaks its parameters iteratively to minimize a given function to its local minimum. This algorithm is widely used in machine learning and deep learning to optimize the parameters W( weights) and b(bias term) for the entire training set.

In this article, we shall learn how the gradient descent algorithm is developed and how it works. Then, using linear regression weights, we shall carry out a python implementation of gradient descent.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
1. Learner will clearly understand how gradient descent work.
2. Later, the learner will be able to develop a gradient descent algorithm independently.
3. Learners will learn how to implement the developed algorithm in no. 2 above; hence they will be able to implement any future algorithm in python.

#### Article quality
This article covers not only the intuition part but also the implementation of our optimization algorithm.

#### References
N/A

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by WanjaMIKE** [comment]
Good afternoon and thank you for submitting your topic suggestion.
Your topic form has been entered into our queue and should be reviewed (for approval) as soon as a content moderator is finished reviewing the ones in the queue before it.
